### DISCUSSION TITLE
Migrating existing userbase

### DISCUSSION THREAD

**Discussion by xumepadismal** [Web3Auth/Web3Auth#402] [Help]
Hi! Our website has an existing userbase with a possibility to login using google, facebook, apple (can be more in future), email+password. There is a possibility for the user to login to the same account using ANY of these login methods, even if they have different emails on e.g. FB and Google accounts. Furthermore, some ofthe users have already their Metamask wallet connected.

Is it possible at all to migrate such a userbase to Web3auth somehow so that single wallet address will be created for one user, whatever login method he/she will use? And if user has already Metamask connected then his/her Metamask wallet address should be preferred.

I'm sorry if it's a messy description but I'm just trying to figure out what options do I have. Thanks!

**Comment by Slyracoon23** [comment]
I am also facing a similar issue as well. From looking at the [docs](https://web3auth.io/docs/troubleshooting/different-wallet-address-issue#while-creating-dapps-with-web3auth-getting-different-wallet-address-after-some-time-while-using-the-same-account-to-login) it seems the anwser is no but I don't know if there is a work-around.

**Comment by himanshuchawla009** [comment]
- You have to use a proxy provider like auth0 which will take of linking user socials by issuing common userId for all social methods.

**Reply by Slyracoon23** [reply]
That's a cool work-around! I will try that solution. 

If you have any other existing code or materials on this solution I would be eternally grateful! @himanshuchawla009

**Reply by himanshuchawla009** [reply]
here is a link that might help you: https://auth0.com/docs/manage-users/user-accounts/user-account-linking

**Reply by AlphaR2** [reply]
Did this work out. I'm trying to find out?
